packages/core/conductor-tasks/bichard_phase_2/processPhase2.ts

```diff
@@ -8,21 +8,32 @@ import { AuditLogEventSource } from "@moj-bichard7/common/types/AuditLogEvent"
 import EventCode from "@moj-bichard7/common/types/EventCode"
 import { isError } from "@moj-bichard7/common/types/Result"
 import CoreAuditLogger from "../../lib/CoreAuditLogger"
+import phase2 from "../../phase2/phase2"
+import pncUpdateDatasetSchema from "../../phase2/schemas/pncUpdateDataset"
+import { Phase2ResultType } from "../../phase2/types/Phase2Result"
 import { unvalidatedHearingOutcomeSchema } from "../../schemas/unvalidatedHearingOutcome"
 import type { AnnotatedHearingOutcome } from "../../types/AnnotatedHearingOutcome"
-import phase2 from "../../phase2/phase2"
+import type { PncUpdateDataset } from "../../types/PncUpdateDataset"
+import { isPncUpdateDataset } from "../../types/PncUpdateDataset"
 
 const s3Config = createS3Config()
 const taskDataBucket = process.env.TASK_DATA_BUCKET_NAME ?? "conductor-task-data"
 const lockKey: string = "lockedByWorkstream"
 
+const schemaFn = (taskData: unknown) =>
+  isPncUpdateDataset(taskData as AnnotatedHearingOutcome) ? pncUpdateDatasetSchema : unvalidatedHearingOutcomeSchema
+
```

Comment:
I have a feeling we should be able to do this without a function - if we created a new schema that was a union of the two schemas it might be able to parse

Reply:
Yes, that works. Thanks! I've updated the code.
I added the new schema in `processPhase2.ts`, but let me know if you think moving it to a separate file would be better.